crates/red_knot_python_semantic/src/types.rs

```diff
@@ -64,8 +64,6 @@ pub fn check_types(db: &dyn Db, file: File) -> TypeCheckDiagnostics {
 
 /// Infer the public type of a symbol (its type as seen from outside its scope).
 fn symbol_by_id<'db>(db: &'db dyn Db, scope: ScopeId<'db>, symbol: ScopedSymbolId) -> Symbol<'db> {
-    let _span = tracing::trace_span!("symbol_by_id", ?symbol).entered();
```

Comment:
I'm inclined to inline the entire `symbol_by_id` function into `symbol` to avoid "missing" symbol lookups in tracing because a symbol gets looked up by its id.